scalyr_agent/monitor_utils/k8s.py

```diff
@@ -2160,21 +2165,30 @@ def __init__(
         self._session.headers.update(headers)
 
         global_log.info("KubeletApi host ip = %s" % self._host_ip)
-        self._kubelet_url = self._build_kubelet_url(kubelet_url_template, host_ip)
+        self._kubelet_url = self._build_kubelet_url(
+            kubelet_url_template, host_ip, node_name
+        )
         self._fallback_kubelet_url = self._build_kubelet_url(
-            FALLBACK_KUBELET_URL_TEMPLATE, host_ip
+            FALLBACK_KUBELET_URL_TEMPLATE, host_ip, node_name
         )
         self._timeout = 20.0
 
     @staticmethod
-    def _build_kubelet_url(kubelet_url, host_ip):
-        if host_ip:
-            return kubelet_url.substitute(host_ip=host_ip)
+    def _build_kubelet_url(kubelet_url, host_ip, node_name):
+        if node_name or host_ip:
```

Comment:
I think this should be an `and`.  I believe the original intention was to refuse to build a URL if you didn't have the host ip.

Reply:
I was thinking of a case where we have one but not the other, we would still be able to build a valid url. But a case where either is missing would be exceptional so and might be better.